### PR TITLE
Update libff to latest version and move binary fields to it

### PR DESCRIPTION
**Pull request by alexander-zw**
Update libff dependency to its latest version, and modify libiop accordingly to be compatible with the new version. All binary fields were moved to libff and there were some API changes for fields and curves. Since libiop also depends on libfqfft, this PR should be merged after the corresponding PR in libfqfft: https://github.com/scipr-lab/libfqfft/pull/17.

Some utils are duplicated in libff and libiop, and the binary field utils are still in libiop. These will be put into an issue for the future.

### CODE REVIEW

File: libiop/tests/snark/test_linking.cpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 #include <gtest/gtest.h>
 
-#include "libiop/algebra/fields/gf64.hpp"
+#include "libff/algebra/fields/binary/gf64.hpp"
```
Comment (ValarDragon) [inline libiop/tests/snark/test_linking.cpp:5]: Needs `<>`

File: libiop/tests/snark/test_ligero_snark.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 #include <libff/algebra/curves/alt_bn128/alt_bn128_pp.hpp>
 
-#include "libiop/algebra/fields/gf64.hpp"
+#include "libff/algebra/fields/binary/gf64.hpp"
```
Comment (ValarDragon) [inline libiop/tests/snark/test_ligero_snark.cpp:7]: Needs `<>`

**Review by ValarDragon — CHANGES_REQUESTED** [review]
LGTM sans import comment

File: libiop/tests/snark/test_ligero_snark.cpp (hunk 2 above)
Comment (alexander-zw) [reply libiop/tests/snark/test_ligero_snark.cpp:7]: Do you mean each binary field? What size?

Comment (ValarDragon) [reply libiop/tests/snark/test_ligero_snark.cpp:7]: Like `find \"libff/algebra/fields/binary/gf64.hpp\"` and `replace <ibff/algebra/fields/binary/gf64.hpp>`, and do so for gf128, gf196, and gf256 as well

**Review by ValarDragon — APPROVED** [review]
LGTM, thanks for upgrading this!

**Pull request merged**